src/shared/svelte-compile-warns/transform/typescript.ts

```diff
@@ -29,7 +29,7 @@ export function transform(
     const output = ts.transpileModule(code, {
       reportDiagnostics: false,
       compilerOptions: {
-        target: ts.ScriptTarget.ESNext,
+        target: context.parserServices.program.getCompilerOptions().target || ts.ScriptTarget.ESNext,
```

Comment:
Thank you for this PR.
Just to be sure, could you please change the following to use optional chaining?

```suggestion
        target: context.parserServices.program?.getCompilerOptions()?.target || ts.ScriptTarget.ESNext,
```

Reply:
Yep, done.